### PR TITLE
[MISC] deprecate views::take

### PR DESCRIPTION
**Pull request by SGSSGene**
solves https://github.com/seqan/product_backlog/issues/330

- Moved file `range/views/take.hpp` → `io/detail/take_view.hpp`
- Moved scope `views::take` → io::detail::take`
- Replaced some `seqan3::views::take` with `std::views::take`

### CODE REVIEW

**Comment by marehr** [comment]
(this PR wins the award for most descriptive title xD)

File: test/performance/range/views/view_take_benchmark.cpp [hunk 1]
```diff
@@ -6,18 +6,19 @@
...
 #include <vector>
 
 #include <benchmark/benchmark.h>
 
+#include <seqan3/io/detail/take_view.hpp>
 #include <seqan3/range/views/single_pass_input.hpp>
-#include <seqan3/range/views/take.hpp>
 #include <seqan3/range/views/take_exactly.hpp>
 
+
```
Comment (Irallia) [inline test/performance/range/views/view_take_benchmark.cpp:21]: Is this newline intended?

**Review by Irallia — APPROVED** [review]
LGFM, thank you!

File: test/performance/range/views/view_take_benchmark.cpp (hunk 1 above)
Comment (SGSSGene) [reply test/performance/range/views/view_take_benchmark.cpp]: Not at all. Thx :-)

File: test/snippet/range/views/deep_with_param.cpp [hunk 2]
```diff
@@ -1,8 +1,8 @@
 #include <vector>
 
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/io/detail/take_view.hpp>
 #include <seqan3/range/views/deep.hpp>
```
Comment (eseiler) [inline test/snippet/range/views/deep_with_param.cpp:5]:
```suggestion
#include <seqan3/alphabet/nucleotide/dna5.hpp>
#include <seqan3/range/views/deep.hpp>
```

File: test/snippet/range/views/take.cpp [hunk 3]
```diff
@@ -1,8 +1,8 @@
+#include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/snippet/range/views/take.cpp:1]: Can you move this snippet to `test/snippet/io/detail/take.cpp`

File: CHANGELOG.md [hunk 4]
```diff
@@ -199,6 +199,8 @@ regression test suite and patches at https://github.com/seqan/seqan3/tree/master
 * We made `seqan3::views::convert` NOAPI and moved it to `seqan3/utility/views/convert.hpp`. You can still use
   `seqan3::views::convert` in the meantime, but we encourage using `std::views::transform` instead as shown in our
   [Cookbook](https://docs.seqan.de/seqan/3.0.3/cookbook.html#cookbook_convert_alphabet_range).
+* We deprecated seqan3::views::take and it will be removed in 3.1.0
```
Comment (eseiler) [inline CHANGELOG.md:202]:
```suggestion
* We deprecated `seqan3::views::take` and it will be removed in 3.1.0. Use `std::views::take` instead
```

**Review by eseiler — CHANGES_REQUESTED** [review]

File: test/snippet/range/views/deep_with_param.cpp (hunk 2 above)
Comment (SGSSGene) [reply test/snippet/range/views/deep_with_param.cpp:5]: I don't know, who or what includes "take_view.hpp", but seems it is used on purpose in this snippet, shouldn't it also be explicit included?

Comment (eseiler) [reply test/snippet/range/views/deep_with_param.cpp:5]: We don't need it because we use `std::views::take`

Comment (eseiler) [reply test/snippet/range/views/deep_with_param.cpp:5]: If you rebase, the snippet changes

Comment (SGSSGene) [reply test/snippet/range/views/deep_with_param.cpp]: hehe, yes I just notice :-) :+1: 
*eseiler is rebasing and compiling 277'000 lines in his head faster than my 8 core machine*

File: include/seqan3/io/detail/take_view.hpp [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+ * | *else*                                                                                        | *implementation defined type*   |
+ *
+ * This adaptor is different from std::views::take in that it performs type erasure for some underlying ranges.
+ * It returns exactly the type specified above.
+ *
+ * Some benchmarks have shown that it is also faster than std::views::take for pure forward and input ranges.
+ *
+ * ### Example
+ *
+ * \include test/snippet/range/views/take.cpp
```
Comment (eseiler) [inline include/seqan3/io/detail/take_view.hpp:614]:
```suggestion
 * \include test/snippet/io/detail/take.cpp
```

**Review by eseiler — APPROVED** [review]

**Comment by marehr** [comment]
@SGSSGene I rebased this for you.

**Pull request merged**